### ISSUE TITLE
Add standing charge?

### ISSUE THREAD

**Issue by NiccyB** [hypercrypt/octopus-db#2]
Hi,
Would it be possible to add the standing charges to the database/totals?
Cheers,
Nicholas.

**Comment by hypercrypt** [comment]
Yes, it's on my list. The next update will be quite a big one, then I'll add Standing Charges

**Comment by klaus-gsma** [comment]
Working on this now. At what level should the standing charge start to be included? 

- Half Hour?
- Hour?
- Day?
- ...

**Comment by hypercrypt** [comment]
Working on this now. At what level should the standing charge start to be included? 

- Half Hour?
- Hour?
- Day?
- ...

**Comment by anothermort** [comment]
From my point of view what is easiest for you to program? 

1. Start of the day i.e. 00:01
2. First Half Hour after midnight.
3. End of Day 
1 and 2 are only important if you are doing a part day roll up of costs.
Which I am not sure is that helpful, as its not like we can get the latest half hour from the API. 
But could be useful to others.

**Comment by hypercrypt** [comment]
If we include standing charge in the HH or Hour, I was going to just add 1/48th or 1/24th of the standing charge to each HH or hour. What I think I'll do if there are no objections is:

- Include 1/48th of the daily standing charge to each half-hour slot (known issue, days when the clock changes will thus be out by just under 1p/day)
- DON'T include the standing charge in the p/kWh calculation as it'll set off @NiccyB's OCD ( https://github.com/hypercrypt/octopus-db/issues/12 )

**Comment by NiccyB** [comment]
I'm not keen on the 1/48th idea - the standing charge applies for the whole day right at the start of the day (i.e. even if you're disconnected half way through the day, you still have to pay the full day's charge for that day). If the standing charge has to be applied to a half hour slot, then I would prefer it's applied to the first half hour slot of the day as this would be the most accurate as this is when the charge becomes due. This also makes it much easier to incorporate variable standing charges if they ever happen in the future (not that I think they will, but it's not beyond the bounds of possibility for Octopus to trial a tariff that has a variable service/standing charge per half-hour slot).
I agree that the figure shouldn't be included in the p/kWh figures - but not for OCD reasons. While it does impact the p/kWh figure in an overview, it's not a variable charge and could/will distort daily figures and trends. If it must be included, then I would prefer for there to be two 'p/kWh' columns, one excluding and one including the standard charge.

**Comment by hypercrypt** [comment]
> latest half hour from the API

If you have Bright set up then you can get up to date data into the DB with `octo bright/import`. Octopus data will overwrite bright data once available.

@NiccyB so should we include standing charge only on the Daily and less granular basis? I might I'll also include an option to turn it off via `octo db/views --no-standing-charge` or similar

**Comment by NiccyB** [comment]
I think, capture the standing charge on a half-hourly basis (as above, at the moment just one charge would apply in the first half hour of the day), but yes, only show it and use it in calculations in daily/weekly/monthly/annual figures.

**Comment by hypercrypt** [comment]
Not sure what you mean by the first bit @NiccyB - about capturing it on ha HH basis?

**Comment by NiccyB** [comment]
I meant that the standing charge should be allocated in the appropriate table in a half hour slot - that is the daily standing charge in the first half hour slot, but with the ability to out a service/standing charge into any half hour slot as required.

**Comment by hypercrypt** [comment]
I am pulling in the standing charge into a new table. The structure for it (API and table) are very similar to the price data. I then just need to pull that into the billing views. If we do not want it showing up in the HH data then there is no "pulling-in" needed here and I can just add it to the daily+ views and not worry about what time the standing charge is charged as it applies per day rather than at a specific time as such.

If people do want it on the HH and H tables then I think I'd prefer it on the HH basis as we're otherwise "loading up" the cost of a particular slot but I think that's a discussion for a new issue in the future unless anyone feels strongly about it

**Comment by NiccyB** [comment]
Sounds good to me.
